docs/src/rsh/consensus/index.md

```diff
@@ -517,7 +517,7 @@ Reach assumes that network tokens and non-network tokens behave identically, but
 
 ---
 
-`{!rsh} Token.burn(tok, amt)`, or `{!rsh} tok.burn(amt)`, where `{!rsh} tok` is a `{!rsh} Token` value and `{!rsh} amt` is a `{!rsh} UInt` value, may be used to @{defn("burn")} tokens in the contract account, meaning that they are utterly destroyed and can never be recovered.
+`{!rsh} Token.burn(tok, amt)`, or `{!rsh} tok.burn(tok)`, where `{!rsh} tok` is a `{!rsh} Token` value and `{!rsh} amt` is a `{!rsh} UInt` value, may be used to @{defn("burn")} tokens in the contract account, meaning that they are utterly destroyed and can never be recovered.  If `{!rsh} amt` is not given, the current balance of the token will be used.
```

Comment:
The first change from `tok` to `amt` seems wrong. Does the code really do that?

Reply:
It takes 1 or 2 args, it always takes a token and optionally takes an amount.  If the amount is not given it uses `getBalanceOf'` on the token.

https://github.com/reach-sh/reach-lang/blob/c2ca1fbc3820cd3641969a7402f9018438b64b69/hs/src/Reach/Eval/Core.hs#L2768

Reply:
Ah, let me try to make a range instead of one line...

https://github.com/reach-sh/reach-lang/blob/c2ca1fbc3820cd3641969a7402f9018438b64b69/hs/src/Reach/Eval/Core.hs#L2768-L2781

Reply:
Note those permalink URLs based on the revision hash *wink* *wink*.

Reply:
Oh, now that I look again, I think you are right -- I didn't notice that it's used as a method call on `tok` instead of as a method call.  So the original was incomplete while my change is wrong.

Reply:
The kicker is this line
https://github.com/reach-sh/reach-lang/blob/c2ca1fbc3820cd3641969a7402f9018438b64b69/hs/src/Reach/Eval/Core.hs#L1397
which makes it so that the first argument is implicitly passed as the variable itself.